applications/RANSModellingApplication/custom_utilities/test_utilities.cpp

```diff
@@ -35,53 +38,70 @@ typedef ModelPart::ElementType ElementType;
 typedef Geometry<NodeType> GeometryType;
 typedef GeometryType::ShapeFunctionsGradientsType ShapeFunctionDerivativesArrayType;
 
-void IsValuesRelativelyNear(const double ValueA, const double ValueB, const double Tolerance)
+/**
+ * @brief based on python's math.IsNear()
+ */
+bool IsNear(const double ValueA, const double ValueB, const double RelTol, const double AbsTol)
 {
-    KRATOS_CHECK_IS_FALSE(!std::isfinite(ValueA));
-    KRATOS_CHECK_IS_FALSE(!std::isfinite(ValueB));
-
-    if (abs(ValueA) < std::numeric_limits<double>::epsilon())
+    if (std::isnan(ValueA) || std::isnan(ValueB))
```

Comment:
are you sure this works as intended? => #369

Reply:
apparently with -ffast-math optimization flag, they do not work, not even isfinite... :/ thanks for pointing this out :)

> why don't you use the check-functionalities from the core?
But in the case of 0==0, you get a warning in this case, which is usefull since i don't want to check 0==0 in my tests (makes more sense to compare some non-trivial values in checks). at the moment core checks doesn't throw a warning message for this :/